health-services/resource-estimation-service/src/main/java/org/egov/processor/util/BoundaryUtil.java

```diff
@@ -0,0 +1,53 @@
+package org.egov.processor.util;
+
+import static org.egov.processor.config.ServiceConstants.HIERARCHYTYPE_REPLACER;
+import static org.egov.processor.config.ServiceConstants.TENANTID_REPLACER;
+
+import org.egov.processor.config.Configuration;
+import org.egov.processor.repository.ServiceRequestRepository;
+import org.egov.processor.web.models.PlanConfigurationRequest;
+import org.egov.processor.web.models.boundary.BoundarySearchResponse;
+import org.egov.tracer.model.CustomException;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Component
+@Slf4j
+public class BoundaryUtil {
+
+	private Configuration config;
+
+	private ServiceRequestRepository serviceRequestRepository;
+
+	private ObjectMapper mapper;
+
+	public BoundaryUtil(Configuration config, ServiceRequestRepository serviceRequestRepository, ObjectMapper mapper) {
+		this.config = config;
+		this.serviceRequestRepository = serviceRequestRepository;
+		this.mapper = mapper;
+	}
+
+	public BoundarySearchResponse search(String tenantId, String hierarchyType,PlanConfigurationRequest planConfigurationRequest) {
+		String boundaryRelationShipSearchLink = getBoundaryRelationShipSearchLink(tenantId, hierarchyType);
+		Object response;
+		BoundarySearchResponse searchResponse = null;
+		try {
+			response = serviceRequestRepository.fetchResult(new StringBuilder(boundaryRelationShipSearchLink),planConfigurationRequest.getRequestInfo());
+			return searchResponse = mapper.convertValue(response, BoundarySearchResponse.class);
+
+		} catch (Exception ex) {
+			log.error("Boundary relationship response error!!", ex);
+			throw new CustomException("BOUNDARY_SEARCH_EXCEPTION", "Failed to parse boundary relationship response for tenantId: "+tenantId);
```

Comment:
Why is there a failure while searching boundary, code not present or response data is not parsable ?

Also is this boundary v2 integration ?

@Priyanka-eGov @devdatta-egov

Reply:
Updated explanation as it will cater for both case.